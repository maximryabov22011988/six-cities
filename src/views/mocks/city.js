export default {
  name: 'Cologne',
  location: [50.938361, 6.959974],
  zoom: 13,
};
