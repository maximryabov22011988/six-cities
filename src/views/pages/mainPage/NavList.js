import React from 'react';
import PropTypes from 'prop-types';

import withActiveItem from '../../hocs/withActiveItem';
import NavItem from './navList/NavItem';

const propTypes = {
  currentCity: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.arrayOf(PropTypes.number),
      zoom: PropTypes.number
    })
  ).isRequired,
  onChangeCity: PropTypes.func.isRequired
};

function NavList({ currentCity, cities, onChangeCity }) {
  return (
    <ul className="locations__list tabs__list">
      {cities.map(city => {
        const WrappedNavItem = withActiveItem(NavItem);
        return (
          <WrappedNavItem
            key={city.name}
            className="tabs__item"
            city={city}
            isActive={city.name === currentCity}
            onChangeCity={onChangeCity}
          />
        );
      })}
    </ul>
  );
}

NavList.propTypes = propTypes;

export default NavList;
