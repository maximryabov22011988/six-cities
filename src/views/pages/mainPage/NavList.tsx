import * as React from 'react';

import NavItem from './navList/NavItem';

import withActiveItem from '../../hocs/withActiveItem';

import { CurrentCity, onChangeCity } from '../../interfaces';

interface Props {
  cities: Array<CurrentCity>,
  currentCity: string,
  onChangeCity: onChangeCity,
}

function NavList({ currentCity, cities, onChangeCity }: Props) {
  return (
    <ul className="locations__list tabs__list">
      {cities.map((city) => {
        const WrappedNavItem = withActiveItem(NavItem);
        return (
          <WrappedNavItem
            city={city}
            className="tabs__item"
            isActive={city.name === currentCity}
            key={city.name}
            onChangeCity={onChangeCity}
          />
        );
      })}
    </ul>
  );
}

export default NavList;
