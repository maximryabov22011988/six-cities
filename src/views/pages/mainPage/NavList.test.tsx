import * as React from 'react';
import * as renderer from 'react-test-renderer';

import NavList from './NavList';
import city from '../../mocks/cities';

const cities = [city];

it('NavList renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <NavList
        cities={cities}
        currentCity="Paris"
        onChangeCity={jest.fn()}
      />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
