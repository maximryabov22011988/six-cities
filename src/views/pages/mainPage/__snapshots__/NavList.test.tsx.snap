// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList renders correctly with mandatory props 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="tabs__item locations__item-link"
      href="#"
    >
      <span>
        Cologne
      </span>
    </a>
  </li>
</ul>
`;
